### ISSUE TITLE
STM32MP157F-DK Build Failing

### ISSUE THREAD

**Issue by blakefuller3** [cvetaevvitaliy/stm32mp1-ubuntu#8]
Hello, I'm building with the STM32MP157F-DK as the target, but the latest commit references the U-Boot submodule which doesn't include DTS files for STM32MP147F (only earlier silicon versions). I see in your latest actions you have successfully built for the xx157F target, but I can't manually update the U-Boot commit version successfully. 

Can you provide any insight in how you successfully targeted xx157F?

**Comment by blakefuller3** [comment]
Hello, I've got the build working now with some slight modifications to the submodule git commits and the uboot and linux scripts. I've attached my raw `git diff`. Perhaps I'm building incorrectly?

My application uses a USB camera and an application running GTK. When I try to execute my application I find that X11 (which GTK defaults to) can't find the display (HDMI or the dev kit DSI display) using the command `xrandr`. I've also tried with WLR-randr but it has the same issue of not being to open the display. Both give the output `Can't open display`, however both DSI and HDMI can output the terminal and respond to `modetest` so I know (at least the phy and driver layer) of the DTS is configured correctly.

Now I have also been able to build the image from your source with the modifications I've posted and have tried a number of attempts to get the display to work, such as building the gcnano library by modifying your script then installing the galcore kernal object, modifying boot args, modifying the DTS to disable DSI and only use HDMI. None of these have have resulted in Wayland or X11 detect the display. 

Can you suggest any reason for this? Thank you for your work in maintaining this repository!

[STM32MP1 Git Diff BFuller 06Feb23.txt](https://github.com/cvetaevvitaliy/stm32mp1-ubuntu/files/10660436/STM32MP1.Git.Diff.BFuller.06Feb23.txt)

**Comment by cvetaevvitaliy** [comment]
For X11 example config file ` /etc/X11/xorg.conf`
```
Section "Device"
 Identifier "ST Accelerated Framebuffer Device"
 Driver "vivante"
 Option "fbdev" "/dev/fb1"
 Option "vivante_fbdev" "/dev/fb1"
 Option "SyncDraw" "false"
EndSection
```

**Comment by blakefuller3** [comment]
Running `xinit` still fails unfortunately. See attached log from X11. I have galcore installed, but I can't see how the X11 server is supposed to find the Vivante kernel object or driver. Below is the output of `lsmod`.

`root@stm32mp1:~# lsmod
Module                  Size  Used by
snd_soc_stm32_sai_sub    28672  1
brcmfmac              221184  0
sha256_generic         16384  0
libsha256              20480  1 sha256_generic
sha256_arm             24576  0
cfg80211              630784  1 brcmfmac
snd_soc_audio_graph_card    20480  0
stusb160x              20480  0
snd_soc_simple_card_utils    20480  1 snd_soc_audio_graph_card
typec                  32768  1 stusb160x
brcmutil               16384  1 brcmfmac
etnaviv                86016  0
gpu_sched              28672  1 etnaviv
hci_uart               36864  0
stm32_crc32            16384  0
btbcm                  16384  1 hci_uart
stm32_cryp             28672  0
stm32_hash             28672  0
snd_soc_cs42l51        24576  0
snd_soc_hdmi_codec     16384  0
bluetooth             491520  3 hci_uart,btbcm
snd_soc_stm32_i2s      20480  0
libdes                 28672  1 stm32_cryp
crypto_engine          16384  2 stm32_hash,stm32_cryp
snd_soc_core          167936  6 snd_soc_stm32_i2s,snd_soc_cs42l51,snd_soc_audio_graph_card,snd_soc_hdmi_codec,snd_soc_simple_card_utils,snd_soc_stm32_sai_sub
snd_pcm_dmaengine      16384  1 snd_soc_core
mousedev               20480  0
snd_pcm                94208  6 snd_soc_stm32_i2s,snd_soc_cs42l51,snd_pcm_dmaengine,snd_soc_hdmi_codec,snd_soc_core,snd_soc_stm32_sai_sub
ecdh_generic           16384  1 bluetooth
ecc                    32768  1 ecdh_generic
snd_timer              32768  1 snd_pcm
libaes                 16384  1 bluetooth
snd                    53248  5 snd_soc_hdmi_codec,snd_timer,snd_soc_core,snd_pcm,snd_soc_stm32_sai_sub
stm32_cec              16384  0
snd_soc_stm32_sai      16384  0
soundcore              16384  1 snd`
[Xorg.0.log](https://github.com/cvetaevvitaliy/stm32mp1-ubuntu/files/11119040/Xorg.0.log)

**Comment by cvetaevvitaliy** [comment]
This is an example of a configuration file.

Pleas refer to the official Xorg documentation

https://www.x.org/wiki/UserDocumentation/

**Comment by blakefuller3** [comment]
I believe the issue is with how galcore is installed. I’ve updated your script build-linux.sh script to build galcore.ko from the gcnano source (the steps are commented out in the source in the repo) and manually installing the kernel object with insmod. Can you clarify how the vivante driver should be built and installed?

**Comment by cvetaevvitaliy** [comment]
Need to apply this patch '00005-add-at-f-stm32mp157a-sodimm2-mx.patch' for kernel

**Comment by blakefuller3** [comment]
Okay I will try that. That patch (and others) were failing during build.

**Comment by cvetaevvitaliy** [comment]
Of course, this patch was made for the specific kernel version specified in the submodule, if you have a different version, you get build error

**Comment by blakefuller3** [comment]
Okay I think this is the issue. To build for -157F I needed to update u-boot and arm-tf submodule to a newer branch as the branch in this repo doesn’t include the DTS and other files for -157F

**Comment by karu2003** [comment]
does not work any more . old patches. :(

**Comment by samaysahu** [comment]
> does not work any more . old patches. :(  

hi did you find the solution for this ? i am also getting the same errro

**Comment by samaysahu** [comment]
> I gave up and switched to yocto

have you build any project using stm32mp157f-dk2 , if yes can you plz tell me what are the things you have build

**Comment by samaysahu** [comment]
i am working on project where i want to use mediapipe library ( python ), but i am not able to download that , so i wanted to know have  you tried downloading any python library on this board ?

**Comment by cvetaevvitaliy** [comment]
> i am working on project where i want to use mediapipe library ( python ), but i am not able to download that , so i wanted to know have you tried downloading any python library on this board ?

I don't have a devboard to debug new distributions and develop the project further. I tried to run the build in CI/CD.
see build log https://github.com/cvetaevvitaliy/stm32mp1-ubuntu/actions/runs/14751616851

**Comment by cvetaevvitaliy** [comment]
You can try Buildroot. it's even easier. Just change the DTS files for your board
https://github.com/cvetaevvitaliy/buildroot-external

**Comment by cvetaevvitaliy** [comment]
@samaysahu 
successful build: https://github.com/cvetaevvitaliy/stm32mp1-ubuntu/actions/runs/14751616851

You can download build artifacts for your board and try it.
